auth0/provider.go

```diff
@@ -77,20 +95,28 @@ func Provider() *schema.Provider {
 }
 
 func Configure(data *schema.ResourceData) (interface{}, error) {
-
-	domain := data.Get("domain").(string)
-	id := data.Get("client_id").(string)
-	secret := data.Get("client_secret").(string)
-	debug := data.Get("debug").(bool)
-
 	userAgent := fmt.Sprintf("Terraform-Provider-Auth0/%s (Go-Auth0-SDK/%s; Terraform-SDK/%s; Terraform/%s)",
 		Version(),
 		SDKVersion(),
 		TerraformSDKVersion(),
 		TerraformVersion())
+	debug := data.Get("debug").(bool)
+	domain := data.Get("domain").(string)
+
+	var authenticationOption management.ManagementOption
+	if token, ok := data.GetOk("auth0_token"); ok {
+		authenticationOption = management.WithStaticToken(token.(string))
+	} else {
+		// if auth0_token is not specified, the user must have provided client_id
+		// and client_sceret due to the constraints in the schema.Provider.schema.
+		// This helps simplying the parsing logic
```

Comment:
Can we simplify this comment to something like:

"If auth0_token is not specified, authenticate with client ID and client secret"